x-pack/plugins/observability_solution/synthetics/server/routes/overview_trends/fetch_trends.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import { SyntheticsEsClient } from '../../lib';
+
+export function fetchTrends(configId: string, locationIds: string[], esClient: SyntheticsEsClient) {
+  const query: SearchRequest = {
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          {
+            bool: {
+              filter: [
+                {
+                  exists: {
+                    field: 'summary',
+                  },
+                },
+                {
+                  bool: {
+                    should: [
+                      {
+                        bool: {
+                          should: [
+                            {
+                              match: {
+                                'summary.final_attempt': true,
+                              },
+                            },
+                          ],
+                          minimum_should_match: 1,
+                        },
+                      },
+                      {
+                        bool: {
+                          must_not: {
+                            bool: {
+                              should: [
+                                {
+                                  exists: {
+                                    field: 'summary.final_attempt',
+                                  },
+                                },
+                              ],
+                              minimum_should_match: 1,
+                            },
+                          },
+                        },
+                      },
+                    ],
+                    minimum_should_match: 1,
+                  },
+                },
```

Comment:
this isn't required anymore, since we are going to fetch at most last 50 checks here, we can resort to just simply filtering for final summary.

Reply:
this is only needed where let's say we are fetching really really old data.

Reply:
👌 6f976f050713181ae0f493ffa17d1885eb5b16d0